tools/graphiteindex/src/main/java/com/criteo/biggraphite/graphiteindex/GraphiteSASI.java

```diff
@@ -0,0 +1,311 @@
+
+        Tracker tracker = baseCfs.getTracker();
+        tracker.subscribe(this);
+
+        SortedSet<SSTableReader> toRebuild = new TreeSet<>(
+            (a, b) -> Integer.compare(a.descriptor.generation, b.descriptor.generation)
+        );
+        for (SSTableReader sstable : tracker.getView().liveSSTables()) {
+            toRebuild.add(sstable);
+        }
+
+        CompactionManager.instance.submitIndexBuild(
+            new GraphiteSASIBuilder(baseCfs, column, toRebuild)
+        );
+    }
+
+    @Override public IndexMetadata getIndexMetadata()
+    {
+        return config;
+    }
+
+    @Override public Callable<?> getInitializationTask()
+    {
+        return null;
+    }
+
+    @Override public Callable<?> getMetadataReloadTask(IndexMetadata indexMetadata)
+    {
+        return null;
+    }
+
+    @Override public void register(IndexRegistry registry)
+    {
+        registry.registerIndex(this);
+    }
+
+    @Override public Optional<ColumnFamilyStore> getBackingTable()
+    {
+        return Optional.empty();
+    }
+
+    @Override public Callable<?> getBlockingFlushTask()
+    {
+        return null;
+    }
+
+    @Override public Callable<?> getInvalidateTask()
+    {
+        return getTruncateTask(FBUtilities.timestampMicros());
+    }
+
+    @Override public Callable<?> getTruncateTask(long truncatedAt)
+    {
+        return () -> {
+            // TODO(d.forest): drop index
+            logger.info("UNIMPLEMENTED - getTruncateTask");
+            return null;
+        };
+    }
+
+    @Override public boolean shouldBuildBlocking()
+    {
+        return true;
+    }
+
+    @Override public boolean dependsOn(ColumnDefinition column)
+    {
+        return this.column.compareTo(column) == 0;
+    }
+
+    @Override public boolean supportsExpression(ColumnDefinition column, Operator op)
+    {
+        // LIKE is the operator that makes the most sense here (pattern-based queries).
+        return dependsOn(column) && (op == Operator.LIKE || op == Operator.LIKE_MATCHES);
+    }
+
+    @Override public AbstractType<?> customExpressionValueType()
+    {
+        return null;
+    }
+
+    @Override public RowFilter getPostIndexQueryFilter(RowFilter filter)
+    {
+        return filter.without(
+            findIndexFilterExpression(filter).orElse(null)
+        );
+    }
+
+    @Override public long getEstimatedResultRows()
+    {
+        // Taken from SASI, makes this index higher-priority for the column it supports.
+        return Long.MIN_VALUE;
+    }
+
+    @Override public void validate(PartitionUpdate update)
+        throws InvalidRequestException
+    {
+    }
+
+    @Override public Indexer indexerFor(
+        DecoratedKey key, PartitionColumns columns, int nowInSec, OpOrder.Group opGroup,
+        IndexTransaction.Type transactionType
+    )
+    {
+        return new Indexer() {
+            @Override public void insertRow(Row row)
+            {
+                // TODO(d.forest)
```

Comment:
can you explain what should be done in these ?

Reply:
done